src/Sylius/Behat/Page/Shipping/ShippingMethodDetailsPage.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Shipping;
+
+use Sylius\Behat\Page\ElementNotFoundException;
+use Sylius\Behat\Page\SymfonyPage;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class ShippingMethodDetailsPage extends SymfonyPage implements ShippingMethodDetailsPageInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getRouteName()
+    {
+        return 'sylius_backend_shipping_method_show';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function pressDelete()
+    {
+        $this->getDocument()->pressButton('Delete');
+
+        $modal = $this->getDocument()->find('css', $modalName = '#confirmation-modal');
+        if (null === $modal) {
+            throw new ElementNotFoundException(sprintf('The element "%s" was not found on page.', $modalName));
+        }
+
+        $confirmButton = $modal->find('css', 'a:contains(Delete)');
+
+        $this->waitForModalToAppear($modal);
```

Comment:
I think it will be helpful if we can decide how long we want to wait for given modal. On different page it can take more time it depends on how complicated modal it will be.